File: ts/Primitives.ts
/**
 * Type syntax, inferrence, const, string handling
 */

let bool: boolean = true && false;
bool = true;
console.log(bool);

let b: number = 0b1010101;
let s: number = 0XFFF;
let i: number = 2147483647;
let l: number = 2147483648;
let c: number = 'a'.charCodeAt(0);
console.log(l - b + c + s + i );

let f: number = 3.14;
let d: number = Math.PI;
console.log(f * d);

let nan: number = NaN;
let inf: number = Infinity;
console.log(nan + inf);

let str: string = 'foo' + 'bar' + c;
console.log(str === 'foobara'); // What's wrong here?

File: java/Primitives.java
public class Primitives {

  public static void main(String[] args) {

    boolean bool = true && false;
    bool = true;
    System.out.println(bool);

    byte b = 0b1010101;
    short s = 0XFFF;
    int i = 2147483647;
    long l = 2147483648L;
    char c = 'a';
    System.out.println(l - b + c + s - i);

    float f = 3.14F;
    double d = Math.PI;
    System.out.println(f * d);

    double nan = Double.NaN;
    double inf = Double.POSITIVE_INFINITY;
    System.out.println(nan + inf);

    String str = "foo" + "bar" + c;
    System.out.println(str == "foobara"); // false??

  }

}
